idaes/models_extra/power_generation/unit_models/cross_flow_heat_exchanger_1D.py

```diff
@@ -0,0 +1,808 @@
+            raise ConfigurationError(
+                "The CrossFlowHeatExchanger1D model is valid only for property packages "
+                "whose single phase is a vapor phase. The hot side phase is not a vapor phase."
+            )
+        if not pobj_cold.is_vapor_phase():
+            raise ConfigurationError(
+                "The CrossFlowHeatExchanger1D model is valid only for property packages "
+                "whose single phase is a vapor phase. The cold side phase is not a vapor phase."
+            )
+
+        # Reference
+        add_object_reference(self, "heat_tube", tube.heat)
+        add_object_reference(self, "heat_shell", shell.heat)
+
+        shell_has_pressure_change = False
+        tube_has_pressure_change = False
+
+        if self.config.cold_side.has_pressure_change:
+            if self.config.shell_is_hot:
+                add_object_reference(self, "deltaP_tube", tube.deltaP)
+                tube_has_pressure_change = True
+            else:
+                add_object_reference(self, "deltaP_shell", shell.deltaP)
+                shell_has_pressure_change = True
+        if self.config.hot_side.has_pressure_change:
+            if self.config.shell_is_hot:
+                add_object_reference(self, "deltaP_shell", shell.deltaP)
+                shell_has_pressure_change = True
+            else:
+                add_object_reference(self, "deltaP_tube", tube.deltaP)
+                tube_has_pressure_change = True
+
+        heat_exchanger_common._make_performance_common(  # pylint: disable=W0212
+            self,
+            shell=shell,
+            shell_units=shell_units,
+            shell_has_pressure_change=shell_has_pressure_change,
+            make_reynolds=True,
+            make_nusselt=True,
+        )
+        heat_exchanger_common._make_performance_tube(  # pylint: disable=W0212
+            self,
+            tube=tube,
+            tube_units=tube_units,
+            tube_has_pressure_change=tube_has_pressure_change,
+            make_reynolds=True,
+            make_nusselt=True,
+        )
+
+        def heat_accumulation_term(b, t, x):
+            return b.heat_accumulation[t, x] if b.config.dynamic else 0
+
+        # Nusselts number
+        @self.Constraint(
+            self.flowsheet().config.time,
+            tube.length_domain,
+            doc="Nusselts number equation on tube side",
+        )
+        def N_Nu_tube_eqn(b, t, x):
+            return (
+                b.N_Nu_tube[t, x]
+                == 0.023
+                * b.N_Re_tube[t, x] ** 0.8
+                * tube.properties[t, x].prandtl_number_phase["Vap"] ** 0.4
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            shell.length_domain,
+            doc="Nusselts number equation on shell side",
+        )
+        def N_Nu_shell_eqn(b, t, x):
+            return (
+                b.N_Nu_shell[t, x]
+                == b.f_arrangement
+                * 0.33
```

Comment:
As above.

Reply:
@jinliangma , do you know where this correlation came from?